op-challenger/cmd/watch/oracle.go

```diff
@@ -0,0 +1,60 @@
+package watch
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/ethereum/go-ethereum/log"
+
+	"github.com/ethereum-optimism/optimism/op-challenger/challenger"
+	"github.com/ethereum-optimism/optimism/op-challenger/config"
+	"github.com/ethereum-optimism/optimism/op-challenger/metrics"
+)
+
+// Oracle listens to the L2OutputOracle for newly proposed outputs.
+func Oracle(logger log.Logger, cfg *config.Config) error {
+	if err := cfg.Check(); err != nil {
+		return fmt.Errorf("invalid config: %w", err)
+	}
+
+	m := metrics.NewMetrics("default")
+
+	service, err := challenger.NewChallenger(*cfg, logger, m)
+	if err != nil {
+		logger.Error("Unable to create the Challenger", "error", err)
+		return err
+	}
+
+	logger.Info("Listening for OutputProposed events from the L2OutputOracle contract", "l2oo", cfg.L2OOAddress.String())
+
+	subscription, err := service.NewOracleSubscription()
```

Comment:
should this have a `defer subscription.Quit()` to ensure the subscription is closed cleanly? The various clients created by challenger aren't closed either.

I don't see where the clients are closed in things like the batcher either so may not matter since the whole process should exit.

Reply:
Agree, I think it makes sense unsubscribe gracefully even though the process exits.